### PR TITLE
Basic ClientHeadersFactory documentation

### LINKED ISSUE #13750: rest-client @RegisterClientHeaders documentation
**Describe the bug**
There should be a clear documentation on how to use RegisterClientHeaders annotation. Actually it's very hard ton find information and I finally got my answers in a microprofile-rest-client issue comment

**Expected behavior**
Add more code samples in extension guides, especially on what can be used as client header factory.

A simple exemple with a cdi bean used as ClientHeadersFactory should be enough.

**Actual behavior**

Only trivial sample code exists now.

If needed and possible I could take care of it.

### PR DESCRIPTION
**Pull request by jtama**
Fixes #13750

### CODE REVIEW

File: docs/src/main/asciidoc/rest-client.adoc [hunk 1]
```diff
@@ -456,6 +456,68 @@ If you use a `CompletionStage`, you would need to call the service's method to r
...
+    Set<Country> getByName(@PathParam String name);
+
+    @GET
+    @Path("/name/{name}")
+    @Produces("application/json")
+    CompletionStage<Set<Country>> getByNameAsync(@PathParam String name);
+}
+----
+
+And the RequestUUIDHeaderFactory` :
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-client.adoc:494]:
```suggestion
And the `RequestUUIDHeaderFactory` would look like:
```

File: docs/src/main/asciidoc/rest-client.adoc [hunk 2]
```diff
@@ -456,6 +456,68 @@ If you use a `CompletionStage`, you would need to call the service's method to r
...
+    @Override
+    public MultivaluedMap<String, String> update(MultivaluedMap<String, String> multivaluedMap, MultivaluedMap<String, String> multivaluedMap1) {
+        MultivaluedMap<String, String> result = new MultivaluedMapImpl<>();
+        result.add("X-request-uuid", UUID.randomUUID().toString());
+        return result;
+    }
+}
+----
+
+Any CDI bean matching the declared factory will be picked-up and can of course benefit from quarkus features.
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-client.adoc:519]:
```suggestion
Any CDI bean matching the declared factory class will be picked up and can of course benefit from the usual CDI infrastructure.
```

File: docs/src/main/asciidoc/rest-client.adoc [hunk 3]
```diff
@@ -456,6 +456,68 @@ If you use a `CompletionStage`, you would need to call the service's method to r
...
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.UUID;
+
+@ApplicationScoped
+public class RequestUUIDHeaderFactory implements ClientHeadersFactory {
+
+    @Override
+    public MultivaluedMap<String, String> update(MultivaluedMap<String, String> multivaluedMap, MultivaluedMap<String, String> multivaluedMap1) {
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-client.adoc:511]:
```suggestion
    public MultivaluedMap<String, String> update(MultivaluedMap<String, String> incomingHeaders, MultivaluedMap<String, String> clientOutgoingHeaders) {
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Very useful addition, thanks!

I added some suggestions.

**Comment by jtama-op** [comment]
I accepted your changes request, I probably need to squash commits now, don't I ?

**Review by gsmet — APPROVED** [review]
I just did the squash :).

Thanks, will merge as soon as CI is green.

**Pull request merged**